magma/common.py

```diff
@@ -205,3 +206,8 @@ def get_child(self, key: str) -> Finalizable:
     def finalize(self):
         for child in self._children.values():
             child.finalize()
+
+
+def lca_of_types(classes):
+    return next(iter(reduce(
+        operator.and_, (collections.Counter(cls.mro()) for cls in classes))))
```

Comment:
Sometimes with code like this I find it useful to leave the link in case any future reader needs to understand it: https://stackoverflow.com/questions/58290137/how-to-find-most-recent-common-ancestor-base-type-of-several-types-in-python